File: src/main/java/com/chrisdempewolf/responses/pin/Pin.java
package com.chrisdempewolf.responses.pin;

public class Pin {
    private final Data data;

    public Pin(Data data) {
        this.data = data;
    }

    public Data getData() {
        return data;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        Pin pin = (Pin) o;

        return !(data != null ? !data.equals(pin.data) : pin.data != null);
    }

    @Override
    public int hashCode() {
        return data != null ? data.hashCode() : 0;
    }

    @Override
    public String toString() {
        return "Pin{" +
                "data=" + data +
                '}';
    }
}

File: src/main/java/com/chrisdempewolf/pinterest/responses/pin/Pin.kt
package com.chrisdempewolf.pinterest.responses.pin

import com.google.gson.annotations.SerializedName

data class Pin(
        val id: String?,
        val counts: Counts?,
        val url: String?,
        val note: String?,
        val link: String?,
        val board: Board?,
        val creator: Creator?,
        @SerializedName("created_at") val createdAt: String?,
        val color: String?,
        val attribution: Attribution?,
        @SerializedName("metadata") val metaData: MetaData?)